### PR TITLE
Project Screen

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #3057 

project sidebar:
- wrapper
- workspace selector
- project list
- project item list
- new project button which open  a modal
- project sidebar footer

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-cli/src/api/resources-create.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ export async function createResource(
   client: ApolloClient<NormalizedCacheObject>,
   name: string,
   description: string,
-  resourceType: models.EnumResourceType
+  resourceType: models.EnumResourceType,
+  projectId: string
 ): Promise<models.Resource> {
```
Comment (yuval-hazaz) [inline packages/amplication-cli/src/api/resources-create.ts:29]: Why is that included in this PR?

File: packages/amplication-cli/src/commands/resources/create.ts [hunk 2]
```diff
@@ -35,21 +35,28 @@ export default class ResourcesCreate extends ConfiguredCommand {
       required: false,
       description: 'description of resource to create',
     },
+    {
+      name: 'projectId',
```
Comment (yuval-hazaz) [inline packages/amplication-cli/src/commands/resources/create.ts:39]: I don't expect this PR to include changes in the CLI

File: packages/amplication-client/src/Layout/InnerTabLink.tsx [hunk 3]
```diff
@@ -8,14 +8,20 @@ type Props = {
   children: React.ReactNode;
   icon: string;
   to: string;
+  onClick?: (selectedItem: any) => void;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Layout/InnerTabLink.tsx:11]: if we decide to use the OnClick event, I think we should use a data object to pass back to the parent. Otherwise, I don't think it will be usable to the parent 

have a look at this for example: 
packages/amplication-design-system/src/components/SelectMenu/SelectMenu.tsx

File: packages/amplication-client/src/Project/AddNewProject.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@import "../style/index.scss";
+
+.add-new-project {
+    &__label {
+        font-size: 12px;
+        line-height: 18px;
+        font-weight: 400;
+    }
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/AddNewProject.scss:8]: please use variable or even better - use @mixin

File: packages/amplication-client/src/Project/AddNewProject.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    <Button
+      onClick={onNewProjectClick}
+      type="button"
+      buttonStyle={EnumButtonStyle.Text}
+      icon="plus"
+      iconPosition={EnumIconPosition.Left}
+      iconSize="xsmall"
+    >
+      <span className={`${CLASS_NAME}__label`}>Add New Project</span>
+    </Button>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/AddNewProject.tsx:26]: This component is just a wrapper to <button> with no additional value. please use <button> instead

File: packages/amplication-client/src/Project/NewProjectWizard.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      <Dialog
+        className="new-entity-dialog"
+        isOpen={newProject}
+        onDismiss={handleNewProjectClick}
+        title="New Project"
+      >
+        <NewProject />
+      </Dialog>
+      <AddNewProject onNewProjectClick={handleNewProjectClick} />
+    </div>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/NewProjectWizard.tsx:27]: instead of deleting "AddNewProject" as suggested above, maybe you should merge it with this one

Comment (yuval-hazaz) [reply packages/amplication-client/src/Project/NewProjectWizard.tsx:27]: I dont like the name "Wizard"
use the name "AddNewProject" of the other component, and include the button inline

File: packages/amplication-client/src/Project/ProjectList.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      border-radius: var(--small-border-radius);
+      padding: 5px 8px;
+      gap: 8px;
+  
+      .amp-icon {
+        font-size: var(--title1-font-size);
+        margin-right: var(--default-spacing-small);
+      }
+    }
+  }
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectList.scss:28]: I don't expect to see any style specific for this component. 
Since you are using the InnerTabLink - all styles should be inherited from it. 
I also talked with @alexbass86 about it - and he expects that all places where we use the InnerTabLink to get the new look as used in the new design

File: packages/amplication-client/src/Project/ProjectListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  project,
+  className,
+  onProjectSelected,
+}: Props) => {
+  return (
+    <InnerTabLink
+      icon="file"
+      to={`/${project.id}`}
+      className={className}
+      onClick={onProjectSelected} // if handleOnClick is needed instead of "to"
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectListItem.tsx:21]: This will not work as it is since InnerTabLink.onClick does not return the project as a parameter

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+  &__inner {
+    height: 688px;
+    width: 248px;
+    border-top: 1px solid #2c3249;
+    border-bottom: 1px solid #2c3249;
+    padding: 16px 0;
+  }
+
+  &__tabs {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:37]: please use variables and mixins

Comment (yuval-hazaz) [reply packages/amplication-client/src/Project/ProjectSideBar.scss:37]: if you need, you can wait to merge with @barshimi before setting the style of this component (until then - just leave it empty)

File: packages/amplication-client/src/Project/ProjectSideBarFooter.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import InnerTabLink from "../Layout/InnerTabLink";
+import "./ProjectSideBarFooter.scss";
+
+const CLASS_NAME = "sidebar-footer ";
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBarFooter.tsx:5]: please use the correct class name based on the component name

File: packages/amplication-client/src/Project/ProjectSideBarFooter.tsx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+const ProjectSideBarFooter = () => {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBarFooter.tsx:7]: I think we should not use "footer" in the name as this component is not directly related to the position where it is delayed. 
Also, it is not related to the project - instead is it related to the workspace. 

I would call it - WorkspaceSettingsNavigation

File: packages/amplication-client/src/Project/ProjectSideBarFooter.scss [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    width: 100%;
+    height: 28px;
+    border-radius: var(--small-border-radius);
+    padding: 5px 8px;
+
+    .amp-icon {
+      font-size: var(--title1-font-size);
+    }
+  }
+}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBarFooter.scss:23]: please use variables and mixin

File: packages/amplication-client/src/Project/ProjectSideBarFooter.scss [hunk 13]
```diff
@@ -0,0 +1,23 @@
+@import "../style/index.scss";
+
+.sidebar-footer {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 0px;
+  gap: 4px;
+  width: 248px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBarFooter.scss:9]: I don't think this component should set it own width

File: packages/amplication-client/src/Project/ProjectSideBarFooter.scss [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  height: 72px;
+  margin: 16px 0;
+
+  .inner-tab-link {
+    width: 100%;
+    height: 28px;
+    border-radius: var(--small-border-radius);
+    padding: 5px 8px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBarFooter.scss:17]: Why are we overriding the style here?

File: packages/amplication-client/src/appRoutes.tsx [hunk 15]
```diff
@@ -268,6 +268,14 @@ export const OldRoutes: RouteDef[] = [
...
     exactPath: false,
   },
+  {
+    path: "/project",
+    Component: lazy(() => import("./Project/ProjectSideBar")),
+    moduleName: "Project",
+    routeTrackType: "project",
+    moduleClass: "project-page",
+    exactPath: true,
+  },
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/appRoutes.tsx:278]: please remove this

File: packages/amplication-prisma-db/prisma/migrations/20220712143206_disconnect_resource_from_workspace/migration.sql [hunk 16]
```diff
@@ -0,0 +1,31 @@
+/*
+  Warnings:
+
```
Comment (yuval-hazaz) [inline packages/amplication-prisma-db/prisma/migrations/20220712143206_disconnect_resource_from_workspace/migration.sql:3]: This should not be included here

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
@abrl91 please see my notes
I didn't review all the changes that are not in amplication-client because I think they are included by mistake.  please remove it from this PR or rebase it so the changes will not be included

**Comment by abrl91** [comment]
> @abrl91 please see my notes I didn't review all the changes that are not in amplication-client because I think they are included by mistake. please remove it from this PR or rebase it so the changes will not be included

I wrote in the description that I checked out from the branch of the disconnection resource from workspace because I thought that I would need to use gql queries and I want to use the updated version of our models and prisma schema

**Comment by yuval-hazaz** [comment]
> checked out from the branch of the disconnection resource from workspace

great. so please change the base of this PR so we can review just the relevant changes

File: packages/amplication-client/src/Layout/InnerTabLink.scss [hunk 17]
```diff
@@ -12,9 +12,11 @@
   justify-content: flex-start;
   margin-bottom: var(--default-spacing-small);
   padding: var(--default-spacing-small);
-  border-radius: var(--default-border-radius);
+  border-radius: var(--small-border-radius);
+  width: 240px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Layout/InnerTabLink.scss:16]: please remove width

File: packages/amplication-client/src/Project/ProjectList.scss [hunk 18]
```diff
@@ -0,0 +1,6 @@
+@import "../style/index.scss";
+
+.project-list {
+  @include flexColumn;
+  margin: 16px 0;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectList.scss:5]: please use variable

File: packages/amplication-client/src/Project/ProjectList.tsx [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+import AddNewProject from "./AddNewProject";
+import "./ProjectList.scss";
+import { ProjectListItem } from "./ProjectListItem";
+
+const CLASS_NAME = "project-list";
+
+type Props = {
+  projects: Project[];
+  HandleProjectChange: (project: Project) => void;
+  onCreateNewProjectClicked: () => void;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectList.tsx:12]: remove this prop

File: packages/amplication-client/src/Project/ProjectListItem.tsx (hunk 8 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Project/ProjectListItem.tsx:18]: @abrl91 what about this?

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 20]
```diff
@@ -0,0 +1,17 @@
+@import "../style/index.scss";
+
+.project-sidebar {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding: var(--large-spacing);
+  gap: var(--large-spacing);
+  width: 280px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:9]: remove width

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+  background: var(--surface);
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:10]: remove

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+  border: 1px solid var(--black10);
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:11]: this should be set on the parent as part of the layout

File: packages/amplication-client/src/style/mixin.scss [hunk 23]
```diff
@@ -125,6 +125,12 @@ $bullet-size: 6px;
   align-items: center;
 }
 
+@mixin flexColumn {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/style/mixin.scss:132]: please make sure this one is required, or do we already have a solution for this use case?

File: packages/amplication-client/src/style/mixin.scss [hunk 24]
```diff
@@ -410,3 +416,11 @@ $active-indicator-width: 4px;
...
   color: #f5f6fa;
 }
+
+@mixin regular-12-gray {
+  @include font-base();
+  font-weight: 400;
+  font-size: 12px;
+  line-height: 18px;
+  color: #a3a8b8;
+}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/style/mixin.scss:426]: please do not add mixin for fonts unless it is part of the bigger initiative to align the texts and fonts in the system

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Project/ProjectListItem.tsx (hunk 8 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/ProjectListItem.tsx:18]: @yuval-hazaz I didn't finish resolving all your comments

File: packages/amplication-client/src/Project/ProjectSideBar.scss (hunk 21 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/ProjectSideBar.scss:10]: remove what? The background? @yuval-hazaz

File: packages/amplication-client/src/style/mixin.scss (hunk 23 above)
Comment (abrl91) [reply packages/amplication-client/src/style/mixin.scss:132]: I searched for these CSS properties before I added this mixin. We don't have something like that, and I assume this is a part of the new design of the sidebars

File: packages/amplication-client/src/Project/ProjectSideBar.scss (hunk 22 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/ProjectSideBar.scss:11]: The parent has a different background color. This sidebar is darker than its parent, and it has its own border and border-radius
image

File: packages/amplication-client/src/Project/ProjectSideBar.scss (hunk 21 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Project/ProjectSideBar.scss:7]: yes, there is no reason to use a background. it should be transparent and get the background from its parent

File: packages/amplication-client/src/Project/ProjectSideBar.scss (hunk 22 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/Project/ProjectSideBar.scss:8]: That should be the parent that sets that for all types of content that will be displayed in it - not just on this page but on all page

File: packages/amplication-client/src/style/mixin.scss (hunk 23 above)
Comment (yuval-hazaz) [reply packages/amplication-client/src/style/mixin.scss:132]: ok

File: packages/amplication-client/src/Layout/InnerTabLink.tsx (hunk 3 above)
Comment (abrl91) [reply packages/amplication-client/src/Layout/InnerTabLink.tsx]: @yuval-hazaz we don't need it

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+.project-sidebar {
+  @include flexColumn;
+  padding: var(--large-spacing);
+  gap: var(--large-spacing);
+  border-radius: var(--default-border-radius);
+  height: 100%;
+
+  &__label {
+    @include regular-12;
+    color: #a3a8b8;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:12]: use var

File: packages/amplication-client/src/Project/ProjectSideBar.scss [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  &__divider {
+    border: 1px solid #2c3249;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.scss:16]: use var

File: packages/amplication-client/src/Project/ProjectSideBar.tsx [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+  currentProject: Project;
+  projectsList: Project[];
+  setNewProject: (project: Project) => void;
+  newProjectRes: () => void;
+  handleSetCurrentWorkspace: (workspaceId: string) => void;
+};
+
+const ProjectSideBar = ({
+  currentWorkspace,
+  currentProject,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.tsx:21]: this is not being used

File: packages/amplication-client/src/Project/ProjectSideBar.tsx [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+  projectsList,
+  setNewProject,
+  newProjectRes,
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectSideBar.tsx:24]: what is this?

File: packages/amplication-client/src/Project/ProjectListItem.tsx [hunk 29]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import InnerTabLink from "../Layout/InnerTabLink";
+import { Project } from "../models";
+
+type Props = {
+  project: Project;
+  onProjectSelected: (project: Project) => void;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/ProjectListItem.tsx:7]: this is not being used

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.scss [hunk 30]
```diff
@@ -0,0 +1,87 @@
...
+    margin-left: 0;
+    margin-right: var(--default-spacing);
+  }
+
+  &__name {
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    font-size: 12px;
+    line-height: 18px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/WorkspaceSelectorNew.scss:30]: use variables or mixin

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.scss [hunk 31]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  &__plan {
+    @include body2;
+    color: var(--black60);
+    font-size: 10px;
+    line-height: 15px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/WorkspaceSelectorNew.scss:37]: use var or mixin

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.tsx [hunk 32]
```diff
@@ -0,0 +1,157 @@
...
+type TSetData = {
+  setCurrentWorkspace: {
+    token: string;
+  };
+};
+
+export const COLOR = "#A787FF";
+const CLASS_NAME = "workspaces-selector-new";
+
+const WorkspaceSelectorNew = () => {
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Project/WorkspaceSelectorNew.tsx:31]: why are we using "new" in the name?

File: packages/amplication-client/src/Project/ProjectSideBar.tsx (hunk 28 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/ProjectSideBar.tsx:24]: These are the props that I suppose to get from the Context

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.tsx (hunk 32 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/WorkspaceSelectorNew.tsx:31]: Because I wasn't sure about it. We already have WorkspaceSerlector component in the Workspace folder. I assume we can replace it, but I didn't want to remove the original one before the review

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.scss (hunk 30 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/WorkspaceSelectorNew.scss:30]: @yuval-hazaz 
I am not sure how to use variables here.
I searched in the project for references, and I saw that that's how we write font size and line height when it is not a part of a mixin. AND none of the mixins we have is precisely what I need.

File: packages/amplication-client/src/Project/WorkspaceSelectorNew.scss (hunk 31 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/WorkspaceSelectorNew.scss:37]: @yuval-hazaz 
I am not sure how to use variables here.
I searched in the project for references, and I saw that that's how we write font size and line height when it is not a part of a mixin. AND none of the mixins we have is precisely what I need.

And another point - these properties and the above from your other comment is copy&paste from our current WorkspaceSelector CSS file

File: packages/amplication-client/src/Workspaces/WorkspaceSelector.scss [hunk 33]
```diff
@@ -21,19 +21,16 @@
...
   &__name {
     white-space: nowrap;
     text-overflow: ellipsis;
     overflow: hidden;
+    font-size: 12px;
   }
 
   &__plan {
-    @include body2;
+    font-size: 10px;
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Workspaces/WorkspaceSelector.scss:33]: please use var

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Workspaces/WorkspaceSelector.scss (hunk 33 above)
Comment (abrl91) [reply packages/amplication-client/src/Workspaces/WorkspaceSelector.scss:34]: we talked about it

File: packages/amplication-client/src/Project/ProjectListItem.tsx (hunk 29 above)
Comment (abrl91) [reply packages/amplication-client/src/Project/ProjectListItem.tsx:7]: Fixed. I don't know why it is not marked as outdated

File: packages/amplication-client/src/Workspaces/WorkspaceSelector.scss [hunk 34]
```diff
@@ -42,19 +39,28 @@
     flex-direction: column;
     align-items: flex-start;
     justify-content: flex-start;
+    font-weight: 400;
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Workspaces/WorkspaceSelector.scss:43]: No mixin that satisfied this rule and keep the style as it should according to Figma
